packages/tools/src/tools/segmentation/strategies/BrushStrategy.ts

```diff
@@ -0,0 +1,289 @@
+import initializeSetValue from './utils/initializeSetValue';
+import initializePreview from './utils/initializePreview';
+import initializeRegionFill from './utils/initializeRegionFill';
+import initializeThreshold from './utils/initializeThreshold';
+import { getStrategyData } from './utils/getStrategyData';
+import { isVolumeSegmentation } from './utils/stackVolumeCheck';
+import type {
+  LabelmapToolOperationDataStack,
+  LabelmapToolOperationDataVolume,
+} from '../../../types/LabelmapToolOperationData';
+
+const { VoxelValue } = utilities;
+export type OperationData =
+  | LabelmapToolOperationDataVolume
+  | LabelmapToolOperationDataStack;
+
+export type InitializedOperationData = OperationData & {
+  // Additional data for performing the strategy
+  enabledElement: Types.IEnabledElement;
+  centerIJK?: Types.Point3;
+  centerWorld: Types.Point3;
+  viewport: Types.IViewport;
+  imageVoxelValue: utilities.VoxelValue<number>;
+  segmentationVoxelValue: utilities.VoxelValue<number>;
+  segmentationImageData: ImageData;
+  previewVoxelValue: utilities.VoxelValue<number>;
+  // The index to use for the preview segment.  Currently always undefined or 255
+  // but define it here for future expansion of LUT tables
+  previewSegmentIndex?: number;
+
+  brushStrategy: BrushStrategy;
+};
+
+export type StrategyFunction = (
+  enabled,
+  operationData: InitializedOperationData
+) => unknown;
+
+export type InitializerInstance = {
+  initDown?: StrategyFunction;
+  completeUp?: StrategyFunction;
+  fill?: StrategyFunction;
+  createInitialized?: StrategyFunction;
+  createIsInThreshold?: StrategyFunction;
+};
+
+export type InitializerFunction = () => InitializerInstance;
+
+export type Initializer = InitializerFunction | InitializerInstance;
+
+/**
+ * Parts to a strategy:
+ * 1. Fill strategy - how the fill gets done (left/right, 3d, paint fill etc)
+ * 2. Set value strategy - can clear values or set them, or something else?
+ * 3. In object strategy - how to tell if a point is contained in the object
+ *    * Bounding box getter for the object strategy
+ * 4. thresholdStrategy - how to determine if a point is within a threshold value
+ *
+ * These combine to form an actual brush:
+ *
+ * Circle - convexFill, defaultSetValue, inEllipse/boundingbox ellipse, empty threshold
+ * Rectangle - - convexFill, defaultSetValue, inRectangle/boundingbox rectangle, empty threshold
+ * might also get parameter values from input,  init for setup of convexFill
+ *
+ * Generate a callback, and a call to pointInShape calling the various callbacks/settings.
+ */
+
+export default class BrushStrategy {
+  /**
+   * Provide some default initializers for various situations, mostly for
+   * external use to allow defining new brushes
+   */
+  public static initializers = {
+    initializePreview,
+    initializeSetValue,
+    initializeThreshold,
+    initializeRegionFill,
+  };
```

Comment:
Do you really need this? I don't see you default to these anywhere

Reply:
Yes, you really need it if you are ever going to implement a strategy externally.  Otherwise, no you don't need it.

Reply:
if there is a strategy externally, then can't we just export those and anyone explicitly use them?